src/smc-webapp/account/actions.ts

```diff
@@ -0,0 +1,354 @@
+        this.setState({ signing_up: false });
+        if (err != null) {
+          // generic error.
+          this.setState({ sign_up_error: { generic: JSON.stringify(err) } });
+          return;
+        }
+        switch (mesg.event) {
+          case "account_creation_failed":
+            this.setState({ sign_up_error: mesg.reason });
+            return;
+          case "signed_in":
+            this.redux.getActions("page").set_active_tab("projects");
+            var { analytics_event, track_conversion } = require("../misc_page");
+            analytics_event("account", "create_account"); // user created an account
+            track_conversion("create_account");
+            return;
+          default:
+          // should never ever happen
+          // alert_message(type:"error", message: "The server responded with invalid message to account creation request: #{JSON.stringify(mesg)}")
+        }
+      }
+    });
+  }
+  // deletes the account and then signs out everywhere
+  delete_account() {
+    async.series(
+      [
+        async cb => {
+          // cancel any subscriptions
+          try {
+            await this.redux.getActions("billing").cancel_everything();
+            return cb();
+          } catch (error) {
+            const err = error;
+            if (this.redux.getStore("billing").get("no_stripe")) {
+              // stripe not configured on backend, so no this err is expected
+              return cb();
+            } else {
+              return cb(err);
+            }
+          }
+        },
+        cb => {
+          // actually request to delete the account
+          return webapp_client.delete_account({
+            account_id: this.redux.getStore("account").get_account_id(),
+            timeout: 40,
+            cb
+          });
+        }
+      ],
+      err => {
+        if (err != null) {
+          this.setState({
+            account_deletion_error: `Error trying to delete the account: ${err}`
+          });
+        } else {
+          this.sign_out(true);
+        }
+      }
+    );
+  }
+
+  forgot_password(email) {
+    return webapp_client.forgot_password({
+      email_address: email,
+      cb: (err, mesg) => {
+        if (mesg != null ? mesg.error : undefined) {
+          err = mesg.error;
+        }
+        if (err != null) {
+          this.setState({
+            forgot_password_error: `Error sending password reset message to ${email} -- ${err}. Write to ${this.help()} for help.`,
+            forgot_password_success: ""
+          });
+          return;
+        } else {
+          this.setState({
+            forgot_password_success: `Password reset message sent to ${email}; if you don't receive it, check your spam folder; if you have further trouble, write to ${this.help()}.`,
+            forgot_password_error: ""
+          });
+          return;
+        }
+      }
+    });
+  }
+
+  reset_password(code, new_password) {
+    return webapp_client.reset_forgot_password({
```

Comment:
return.  Fix all of them.  It makes me nervous (in light of some issues above) that you haven't even read over the autogenerated code.  I am all for doing a commit with the autogenerated code, but I would hope for some minimal actually reading over it and fixing it up before merging a PR...